jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/CustomRequestLogTest.java

```diff
@@ -220,209 +246,301 @@ public void testLogAddress() throws Exception
-
-        _connector.getResponse("GET / HTTP/1.0\n\n");
-        String log = _entries.poll(5, TimeUnit.SECONDS);
+        start("EnvironmentVar: %{JAVA_HOME}e");
 
+        HttpTester.Response response = getResponse("GET / HTTP/1.0\n\n");
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        String log = _logs.poll(5, TimeUnit.SECONDS);
         String envVar = System.getenv("JAVA_HOME");
         assertThat(log, is("EnvironmentVar: " + ((envVar == null) ? "-" : envVar)));
     }
 
     @Test
     public void testLogRequestProtocol() throws Exception
     {
-        testHandlerServerStart("%H");
+        start("%H");
 
-        _connector.getResponse("GET / HTTP/1.0\n\n");
-        String log = _entries.poll(5, TimeUnit.SECONDS);
+        HttpTester.Response response = getResponse("GET / HTTP/1.0\n\n");
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        String log = _logs.poll(5, TimeUnit.SECONDS);
         assertThat(log, is("HTTP/1.0"));
     }
 
     @Test
     public void testLogRequestHeader() throws Exception
     {
-        testHandlerServerStart("RequestHeader: %{Header1}i, %{Header2}i, %{Header3}i");
+        start("RequestHeader: %{Header1}i, %{Header2}i, %{Header3}i");
 
-        _connector.getResponse("GET / HTTP/1.0\nHeader1: value1\nHeader2: value2\n\n");
-        String log = _entries.poll(5, TimeUnit.SECONDS);
+        HttpTester.Response response = getResponse("""
+            GET / HTTP/1.0
+            Header1: value1
+            Header2: value2
+
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        String log = _logs.poll(5, TimeUnit.SECONDS);
         assertThat(log, is("RequestHeader: value1, value2, -"));
     }
 
     @Test
     public void testLogKeepAliveRequests() throws Exception
     {
-        testHandlerServerStart("KeepAliveRequests: %k");
+        start("KeepAliveRequests: %k");
 
-        LocalConnector.LocalEndPoint connect = _connector.connect();
-        connect.addInput("""
+        getResponses("""
             GET /a HTTP/1.0
             Connection: keep-alive
 
-            """);
-        connect.addInput("""
             GET /a HTTP/1.1
             Host: localhost
+                        
+            GET /a HTTP/1.0
+                        
+            """, 3);
 
-            """);
-
-        assertThat(connect.getResponse(), containsString("200 OK"));
-        assertThat(connect.getResponse(), containsString("200 OK"));
-
-        connect.addInput("GET /a HTTP/1.0\n\n");
-        assertThat(connect.getResponse(), containsString("200 OK"));
-
-        assertThat(_entries.poll(5, TimeUnit.SECONDS), is("KeepAliveRequests: 1"));
-        assertThat(_entries.poll(5, TimeUnit.SECONDS), is("KeepAliveRequests: 2"));
-        assertThat(_entries.poll(5, TimeUnit.SECONDS), is("KeepAliveRequests: 3"));
-    }
-
-    @Disabled
-    @Test
-    public void testLogKeepAliveRequestsHttp2() throws Exception
```

Comment:
Correct, because we are in `jetty-server` we cannot have the http2 dependencies.
I have no idea why this test was added in the first place, but it's disabled in every Jetty version for the reasons above.

Reply:
Fair enough